python/pyiceberg/io/pyarrow.py

```diff
@@ -409,22 +418,22 @@ def visit_not_null(self, term: BoundTerm[Any]) -> pc.Expression:
         return pc.field(term.ref().field.name).is_valid()
 
     def visit_equal(self, term: BoundTerm[Any], literal: Literal[Any]) -> pc.Expression:
-        return pc.field(term.ref().field.name) == literal.value
+        return pc.field(term.ref().field.name) == _convert_scalar(literal.value, term.ref().field.field_type)
```

Comment:
How does this know what timestamp format our filter is in? Does it assume micros?

Reply:
We explicitly tell it to read `micros`:

```python
@_iceberg_to_pyarrow_type.register
def _(_: TimestampType) -> pa.DataType:
    return pa.timestamp(unit="us")

@_iceberg_to_pyarrow_type.register
def _(_: TimestamptzType) -> pa.DataType:
    return pa.timestamp(unit="us", tz="+00:00")
```
https://github.com/apache/iceberg/blob/master/python/pyiceberg/io/pyarrow.py#L369-L376

It also looks like PyArrow is doing the conversion from seconds and milliseconds to microseconds: https://github.com/apache/iceberg/pull/6070 We don't support nanoseconds because then we would lose precision.